tests/functional/test_configured_endpoint_url.py

```diff
@@ -0,0 +1,279 @@
+        'profile': 'default',
+        'config_file_contents': dict_to_ini_section(
+            {
+                'aws_access_key_id': '123',
+                'aws_secret_access_key': '456',
+                'region': 'fake-region-10',
+            },
+            "profile default",
+        ),
+        'environment': {
+            f'AWS_ENDPOINT_URL_{transformed_service_id}': 'https://endpoint-override'
+        },
+        'client_args': {},
+    }
+
+
+def create_service_config_test_case(service_name, service_model):
+    transformed_service_id = service_model.service_id.replace(' ', '_').lower()
+
+    profile_section = dict_to_ini_section(
+        {
+            'services': 'my-services',
+            'aws_access_key_id': '123',
+            'aws_secret_access_key': '456',
+            'region': 'fake-region-10',
+        },
+        "profile default",
+    )
+
+    services_section = dict_to_ini_section(
+        {
+            transformed_service_id: {
+                'endpoint_url': 'https://endpoint-override'
+            }
+        },
+        "services my-services",
+    )
+
+    return {
+        'service': service_name,
+        'profile': 'default',
+        'config_file_contents': profile_section + services_section,
+        'environment': {},
+        'client_args': {},
+    }
+
+
+SERVICE_TO_OPERATION = {
+    'acm-pca': {'operation': 'list_certificate_authorities', 'params': {}},
+    'apigateway': {'operation': 'get_account', 'params': {}},
+    'cloudfront': {'operation': 'list_distributions', 'params': {}},
+    'cloudtrail': {'operation': 'list_trails', 'params': {}},
+    'codebuild': {'operation': 'list_projects', 'params': {}},
+    # 'ec2': {'operation': 'describe_instances', 'params': {}},
+    's3': {'operation': 'list_buckets', 'params': {}},
+    # 'sts': {'operation': 'get_caller_identity', 'params': {}},
+}
+
+
+def create_service_to_operation_cases():
+    available_services = [
+        x
+        for x in _known_service_names_and_models()
+        if x[0] in SERVICE_TO_OPERATION.keys()
+    ]
+
+    for x in available_services:
+        yield pytest.param(x[0], x[1], SERVICE_TO_OPERATION[x[0]])
+
+
+@pytest.mark.parametrize("test_case", create_cases())
+def test_resolve_configured_endpoint_url(test_case, client_creator):
+    client = client_creator(
+        service=test_case['service'],
+        profile=test_case['profile'],
+        client_args=test_case['client_args'],
+        config_file_contents=test_case['config_file_contents'],
+        environment=test_case['environment'],
+    )
+
+    assert_client_endpoint_url(
+        client=client,
+        expected_endpoint_url=test_case['expected_endpoint_url'],
+    )
+
+
+@pytest.mark.parametrize(
+    'service_name,service_model', _known_service_names_and_models()
+)
+def test_service_env_var_name_is_correct(
+    service_name, service_model, client_creator
+):
+    test_case = create_service_env_test_case(service_name, service_model)
```

Comment:
I have the same comment as the shared config version of this test. Let's remove this helper test function and have the inputs to `create_client()` explicitly defined in the test case body to make the test easier to follow.